packages/matter.js/src/codec/MessageCodec.ts

```diff
@@ -13,6 +13,9 @@ import { DiagnosticDictionary } from "../log/Logger.js";
 export interface PacketHeader {
     sessionId: number,
     sessionType: SessionType,
+    hasPrivacyEnhancements: boolean,
+    isControlMessage: boolean,
+    hasMessageExtensions: boolean,
```

Comment:
If I'm reading correctly these aren't actually used anywhere because we throw rather than set them.  Maybe you could set them in MessageCodec and throw the "unsupported" message at a higher level?

Reply:
Yeah I think they're good to have in.  Since you already retrieve the values (but then immediately throw) I was thinking you could instead set them in PacketHeader and have whoever reads PacketHeader do the throw instead.  NBD though